client/network/src/request_responses.rs

```diff
@@ -584,7 +594,9 @@ impl NetworkBehaviour for RequestResponsesBehaviour {
 							return Poll::Ready(NetworkBehaviourAction::GenerateEvent(out));
 						}
 						RequestResponseEvent::ResponseSent { request_id, peer } => {
-							let arrival_time = self.pending_responses_arrival_time.remove(&request_id)
+							let arrival_time = self.pending_responses_arrival_time.remove(
+								&(protocol.clone(), request_id),
+							)
```

Comment:
The expect below should be using a correct proof, as it is required in substrate.

Reply:
Altered the proof with https://github.com/paritytech/substrate/pull/7957/commits/674ca6c53356fa6347089cd74dad6b1fdae4fec9. @bkchr is this enough to be considered a _correct proof_ in Substrate?

Reply:
Yes, ty.